plugins/rbac/src/components/CreateRole/CreateRoleForm.test.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+
+import { CreateRoleForm } from './CreateRoleForm';
+
+jest.mock('react-router-dom', () => ({
+  useNavigate: jest.fn(),
+}));
+
+jest.mock('@backstage/core-plugin-api', () => ({
+  ...jest.requireActual('@backstage/core-plugin-api'),
+  useApi: jest.fn(),
+}));
+
+jest.mock('formik', () => ({
+  ...jest.requireActual('formik'),
+  useFormik: jest.fn().mockReturnValue({
+    errors: {},
+    values: {},
+    status: { submitError: 'Unable to create role' },
+  }),
+}));
+
+describe('CreateRoleForm', () => {
+  it('renders create role form correctly', async () => {
+    render(<CreateRoleForm />);
+
+    expect(
+      screen.getByText(/enter name and description of role/i),
+    ).toBeInTheDocument();
+    expect(screen.getByTestId(/role-name/i)).toBeInTheDocument();
+    expect(screen.getByTestId(/role-description/i)).toBeInTheDocument();
+    expect(screen.getByText(/add members/i)).toBeInTheDocument();
+  });
+
+  it('handles form submission failure', async () => {
+    render(<CreateRoleForm />);
+
+    expect(screen.getByText(/unable to create role/i)).toBeInTheDocument();
```

Comment:
@gashcrumb On submission if there is any error, formik status gets updated with that error.
```
{formik.status?.submitError && (
      <Box>
          <Alert severity="error">{`Unable to create role. ${formik.status.submitError}`}</Alert>
       </Box>
  )}
```
So here I was just trying to test if any such error is present in the status, the error message is displayed or not. I have mocked formik and returning the status with an error to test the above condition.  I will update the test so that its not confusing.